src/components/map-view/MapView.tsx

```diff
@@ -539,8 +552,12 @@ export const MapView = memo(
         };
       }, []);
 
+      const transformedLightStyle = useMemo(() => {
+        return props.light ? transformStyle(props.light) : undefined;
+      }, [props.light]);
```

Comment:
I think this can be within the other `useMemo`, there is no benefit of separating as the other subscribes to `props` anyways.